[do-not-merge]🎨 Add code of conduct references and template

rmarting opened the pull request:
This is a proposal to extend the Standard Base Documentation pattern adding references about the Code of Conduct of an InnerSource project. This PR is to complete the issue #555.

Ready for a review for the community and get feedback.

yuhattor committed:
Initial Japanese Translation - extensions-for-sustainable-growth pattern

yuhattor committed:
Update links

yuhattor committed:
Update the wiki link to pass the link check

spier committed:
Opting for a shorter name for the GHA

spier committed:
Upgrade toe checkout@v3, to prevent warning 'Node.js 12 actions are deprecated.'

spier committed:
Some formatting of comments and spacing, to improve readability

spier committed:
Document when the cron runs

spier committed:
Add example of where we might test for the existence of the issue.md file

yuhattor committed:
Update translation/ja/patterns/extensions-for-sustainable-growth.md

Co-authored-by: tomoyuki kobayashi <kobatommy@gmail.com>

yuhattor committed:
Update translation/ja/patterns/extensions-for-sustainable-growth.md

Co-authored-by: tomoyuki kobayashi <kobatommy@gmail.com>

yuhattor committed:
Update translation/ja/patterns/extensions-for-sustainable-growth.md

LGTM です!

Co-authored-by: tomoyuki kobayashi <kobatommy@gmail.com>

yuhattor committed:
Update translation/ja/patterns/extensions-for-sustainable-growth.md

Co-authored-by: tomoyuki kobayashi <kobatommy@gmail.com>

yuhattor committed:
Update translation/ja/patterns/extensions-for-sustainable-growth.md

Co-authored-by: tomoyuki kobayashi <kobatommy@gmail.com>

yuhattor committed:
Incorporated the review

Co-authored-by: bakisunsan <bakisunsan@users.noreply.github.com>

yuhattor committed:
Update Japanese translation

rmarting committed:
🎨 Add code of conduct references and template

rmarting committed:
🎨 Add code of conduct references and template

spier commented on patterns/2-structured/project-setup/base-documentation.md:
```diff
@@ -81,6 +82,18 @@ topics:
+helps prevent and address issues such as harassment, discrimination, and toxic behavior, ensuring that
+everyone feels safe and valued within the project. By providing a framework for positive and inclusive
+participation, a Code of Conduct encourages diverse perspectives, enhances collaboration, and ultimately
+leads to the development of higher-quality open-source software.
```
```suggestion
leads to the development of higher-quality InnerSource software.
```

rmarting committed:
💚 Update broken urls from Storyset

rmarting committed:
✏️ Remove trailing spaces

rmarting committed:
💚 Link to the correct file

kschueths committed:
Create COMMUNICATION-template.md (#557)

Create COMMUNICATION-template.md - Creating template so I have link to add to the base documentation pattern when I add the template to it.

---------

Co-authored-by: Sebastian Spier <github@spier.hu>

rrrutledge committed:
Add Known Instance to service-vs-library.md (#559)

yuhattor committed:
Update translation/ja/patterns/extensions-for-sustainable-growth.md

Co-authored-by: tomoyuki kobayashi <kobatommy@gmail.com>

yuhattor committed:
Update translation/ja/patterns/extensions-for-sustainable-growth.md

Co-authored-by: tomoyuki kobayashi <kobatommy@gmail.com>

spier committed:
Add Mercedes-Benz as known instance to innersource-portal.md (#561)

spier committed:
Use glob to select files to be checked by vale (#541)

* Use glob to select files to be checked by vale
* Removing GH token
* Remove unused gitignore rule
* Adding debug flag, to increase inspectability of vale output in future GHA runs

dependabot[bot] committed:
Bump semver from 5.7.1 to 5.7.2 in /pattern-categorization (#558)

Bumps [semver](https://github.com/npm/node-semver) from 5.7.1 to 5.7.2.
- [Release notes](https://github.com/npm/node-semver/releases)
- [Changelog](https://github.com/npm/node-semver/blob/v5.7.2/CHANGELOG.md)
- [Commits](https://github.com/npm/node-semver/compare/v5.7.1...v5.7.2)

---
updated-dependencies:
- dependency-name: semver
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

spier commented on patterns/2-structured/project-setup/base-documentation.md:
```diff
@@ -81,6 +82,18 @@ topics:
 
+### CODE_OF_CONDUCT.md
+
+Defining a Code of Conduct within an InnerSource project is essential for fostering a welcoming and
```
```suggestion
Are open collaboration and sharing really new for your organization? Does your organization consists of teams with vastly different team cultures (often happens through organization growth and acquisitions)?

In these cases defining a Code of Conduct within your InnerSource project will help to foster a welcoming and
```

The previous section for the CONTRIBUTING.md contains a qualifier, stating when a separate file makes sense:
> If the explanation of the steps to make a contribution are too involved,

I wonder if we could add something similar to the beginning of the CODE_OF_CONDUCT.md section?

I think we could use bits and pieces from this comment of yours:

> From my field experience, CoC helps to create a safe space of collaboration when the community is starting inside an organization where open collaboration and sharing can be really new. It is very common in organizations with some kind of "common culture", but it is also common that each team has an specific way of working, processes, or behaviors. That "team culture" (sometimes we identify as a ["Social Contract"](https://openpracticelibrary.com/practice/social-contract/)) is created by the team, and when they are opening the project, they need to share how the community should work, how the communications are expected, and how the conversations must be addressed.
> 
> If you want to success extending the social contract of the team to the rest of the organization, the CoC is a good resource. It is important describe how you are expecting your contributors, users, and community members should speak each others.
> 

I am leaving a proposal in the suggestion to help you understand the way that I am thinking about this. Feel free to change this entirely :)

Also if we could somehow add the ["Social Contract"](https://openpracticelibrary.com/practice/social-contract/) concept somewhere (including the link) that would be great. Maybe it is a useful exercise for a team to do, before they write their CoC?

spier committed:
Adding Acknowledgments section to attribute the additional contributors to this pattern

spier committed:
Change the order of acknowledgements

spier commented on patterns/2-structured/project-setup/base-documentation.md:
```diff
@@ -115,6 +133,11 @@ started right away: [README-template.md](templates/README-template.md) and
 
 * Isabel Drost-Fromm
 
+## Acknowledgments
```
FYI I added this Acknowledgements section, to give praise where due :)

spier commented on patterns/2-structured/project-setup/templates/CODE_OF_CONDUCT-template.md:
```diff
@@ -0,0 +1,78 @@
+<!--
```
Great idea with the variables here. Nice!

spier commented on patterns/2-structured/project-setup/templates/CODE_OF_CONDUCT-template.md:
```diff
@@ -0,0 +1,78 @@
+- {{ CommunityMail-Link }}
+-->
+
+# Code of Conduct
```
You mention that this template is adapted from https://www.contributor-covenant.org/version/2/1/code_of_conduct/.

Could you add some context to the major changes you made, so that I can understand why they were made?

The changes I spotted:

- left away the 2nd sentence of pledge 
> "We pledge to act and interact in ways that contribute to an open, welcoming, diverse, inclusive, and healthy community."
- renamed "Enforcement Responsibilities" to "Our Responsibilities"
- left away the "Enforcement Guidelines"

I am sure I missed other changes that were made :)

spier commented on patterns/2-structured/project-setup/templates/CODE_OF_CONDUCT-template.md:
```diff
@@ -0,0 +1,78 @@
+
+## Our Pledge
+
+In the interest of fostering an open and welcoming environment, {{ Community-Name }} as
```
Will this sentence read correctly when the placeholder is replaced?

I tried this with the example of our own patterns community:

> In the interest of fostering an open and welcoming environment, the **InnerSource Patterns Community** as contributor and maintainer pledge to making participation in this project a harassment-free experience for everyone, ...

I think it is the "as contributor and maintainer" that is throwing me off a bit. Not quite sure.

spier committed:
Adding CoC template to the menu in the rendered book. Note: Translations of this template to other languages would have to happen later.

spier commented on assets/img/standard-base-documentation/README.md:
```diff
@@ -4,8 +4,10 @@ The current illustration is a digital remake of this [original visual](/patterns
 If you want to edit this illustration, please request access to this [source document](https://docs.google.com/presentation/d/11JOByEO9QXlRLXX5BIv9rjUzPzCKErZzknD1OLcprQQ/edit?usp=sharing).
```
Do you have an email address that I can use to add you as Editor to this [Google Doc](https://docs.google.com/presentation/d/11JOByEO9QXlRLXX5BIv9rjUzPzCKErZzknD1OLcprQQ/edit#slide=id.g15a686b4ea4_0_70)?

Then we could add your illustration there, to make it easier to edit it in the future.

Also I might try to change formatting a tiny bit, to make it even more consistent with the other two visuals that we already have. (What I have in mind are  coloring and centering the text below the image).

spier committed:
We are adding another template now. Simplifying the text to prevent repetition and make it easier to extend in the future.

spier committed:
Adding a new sub-section to separate the explanation of the individual files from general information about how to get started with writing these files

spier committed:
Adding a new sub-section to separate the explanation of the individual files from general information about how to get started with writing these files

spier reviewed (changes requested):
Again great work on this PR @rmarting.

I left various comments inline. If you could go through them and comment/resolve/reject as you see fit, that would be great.

I think there are only one or two places that require a bit more work but then we should be good to get this PR merged 👍 

Looking forward to your feedback.

spier committed:
Adding welcome bot + fix links to slack (#567)

* Adding configuration for the welcome bot.
* Update link to the ISC slack instance.
* Stop checking link to chatham house rule (as it always gets reported as broken, even though it is not).

dgrizzanti committed:
Move Release Process pattern to L2 (Structured) (#524)

Move Release Process to structured by doing various changes:

* Update title
* Update problem category (in the mindmap). Also added Standard Base Documentation to this new problem category.
* Updates to Context and Force. Plus a few grammar clean ups
* Update Patlet
* Restructure Problem and Context

spier committed:
Writing updated files for the book

zkoppert committed:
Add GitHub to release-process as a known instance (#568)

spier committed:
Re-creating markmap and screenshot

spier committed:
Remove outdated link to 'release process' pattern. this has been upgraded to Structured. (#574)

spier committed:
Porting board report for this WG here (#576)

Merging without review, as the review happens on a different repo.

spier committed:
Change description of available languages, so that we don't have to change it when adding further langauges.

yuhattor committed:
Updating Japanese🇯🇵 / Chinese🇨🇳  page to the latest (#548)

* Updated Japanese translation - innersource-portal.md
* Updated Japanese translation - maturity-model.md
* Checked Japanese translation - contracted-contributor.md
* Update translation history
* Checked Japanese translation - document-your-guiding-principles.md
* Japanese translation - Checked files that do not need to be updated and update the update date
* Check the i18n 🇨🇳 consistency
* Remove individual names from consistency check-only update information
* Inserted new-line to pass the lint checker
* Update the updated date

yuhattor committed:
Refactor workflow for i18n consistency check and make it work properly again

* Refactor code

* FIx bugs

* Update token

* Update scripts

* Small fix

spier committed:
Host translations in gitbook, using monorepo approach (#578)

* Remove the main gitbook yaml file. Need to be sure that config is picked up from subfolders.
* Change config of the en book
* Change config of the ja book
* Change config of the zh book
* Generate the TOCs for all books in a GHA matrix build
* Change commit messsage in book.yaml workflow run to make clear which file was changed and why
* Improve documentation

spier committed:
[docs] Fix link

spier committed:
New formatting and modified content for newPRWelcomeComment (welcome bot)

yuhattor committed:
Update .github/workflows/i18n-consistency-checker.yaml

LGTM! thanks

Co-authored-by: Sebastian Spier <github@spier.hu>

yuhattor committed:
Update .github/workflows/i18n-consistency-checker.yaml

Co-authored-by: Sebastian Spier <github@spier.hu>

yuhattor committed:
Update .github/workflows/i18n-consistency-checker.yaml

yuhattor committed:
Update .github/workflows/i18n-consistency-checker.yaml

LGTM!

Co-authored-by: Sebastian Spier <github@spier.hu>

yuhattor committed:
Update .github/workflows/i18n-consistency-checker.yaml

yuhattor committed:
Update title extraction

yuhattor committed:
deleting space

jrcosta committed:
InnerSource patterns book translation into Brazilian Portuguese (#577)

Initial translation of patterns to into Brazilian Portuguese
* Using pt-b as translation folder name
* Putting all translation authors names in patterns
* Adding CODEOWNERS for pt-br language. Also fixing the same for previous languages (ja/zh).
* Adapt config for pt-br book to monorepo approach

---------

Co-authored-by: jrcosta <jrcosta@users.noreply.github.com>
Co-authored-by: Humberto Zilio <hzilio@gmail.com>
Co-authored-by: DanielleAlmeida <DanielleAlmeida@users.noreply.github.com>
Co-authored-by: Sebastian Spier <github@spier.hu>

spier committed:
Run generation script for toc.md on correct folder (pt-br)

spier committed:
Render RFC templates in book (#585)

* Add RFC template to menu of the book. That way it renders in gitbook, rather than redirecting to github.
* Adding zh translation of RFC template to the book as well

kschueths committed:
Update innersource-customer-interview-questions.md (#588)

Added ADI as known instance

spier committed:
Fixing link fragments in translations (#590)

Fixing issues with link fragments (MD051) in all translations.

kschueths committed:
Adding the COMMUNICATION.md to the Standard Base Documentation pattern (#589)

* Added COMMUNICATION.md and ADI instance to base-documentation.md.
* Adding COMMUNICATION template to the sidebar of the book (en only)0

spier committed:
Writing updated toc.md for the en book

spier committed:
Removed unused comments from CODEOWNERS

spier committed:
Improved Trusted Committer documentation (#595)

[docs] Add information about Translation Leads + other minor fixes

zkoppert committed:
Create pattern-metrics.yaml (#593)

* Create pattern-metrics.yaml

Getting started measuring pattern metrics. Related to #565. 

This pull request implements
- Time to first response metrics for new issues (excluding codeowners created) recurring every month
- Time to first response metrics for new pull requests (excluding codeowners created) recurring every month

spier committed:
Fix broken indentation for YAML file

spier committed:
Rename GHA - we will add more than just issue metrics

spier committed:
[GHA issue metrics] Make the codeowners uniq (#598)

* Make the list of CODEOWNERS uniq. Mostly a cosmetic change to make the SEARCH_QUERY shorter
* Use sparse-checkout for the 2nd job as well

spier committed:
Fix URLs of images used by the WelcomeBot (#600)

zkoppert committed:
pattern-metrics: Merge to one report (#599)

* Merge the 2 pattern-metrics reports into a single report
* Change title of created issue
* Add 'community metrics' label to created issue

---------

Co-authored-by: Sebastian Spier <github@spier.hu>

spier committed:
Fix YAML

spier committed:
Fix Yaml (quote title of issue to enable us of colon)

spier committed:
Add monthly contributors report (#602)

* Add workflow that will create a monthly report of contributors to this repo

spier committed:
Adding start date and end date to the title of the monthly contributors report

Commit:
Translation of new InnerSource Patterns into Brazilian Portuguese (#606)

* Fixing dates on translated patterns
* Translating pattern extensions-for-sustainable-growth.md to pt-br
* Translating pattern group-support.md to pt-br
* Translating pattern release-process.md to pt-br

---------

Co-authored-by: Sebastian Spier <github@spier.hu>

spier committed:
Re-generating ToC for pt-br translations (#607)

* Allow for GHA for gitbook generation to be run manually
* Writing updated toc.md for the pt-br book

zkoppert committed:
Switch to pull_request_target strategy (#608)

fixes #581

zkoppert committed:
Reconfigure branch specifications with pull_request target (#609)

Related to https://github.com/InnerSourceCommons/InnerSourcePatterns/issues/581

spier committed:
Rename step in GHA (cosmetics)

psanxiao committed:
Add new translation language: Galician (#613)

* Added Galician translation (incl. some translated images, mindmap, and pattern template)
* Added @psanxiao as translation lead for Galician

---------

Co-authored-by: Sebastian Spier <github@spier.hu>

spier committed:
Automate the generation of the "Explore Platterns" mindmap for multiple languages (#618)

Using a matrix build of the GHA to generate the "Explore Platterns" mindmap for multiple languages in parallel.
Also upgrade dependencies (to apply security fixes).

spier committed:
Add pt-br and gl to the translation consistency check

spier committed:
Add maintainer section, related to translations

spier committed:
Explain how to regenerate the mindmap for different languages

spier committed:
Add basic documentation about translating visuals. extend mainter docs.

spier committed:
Remove duplicated files. Script is run from parent folder.

tsadler1988 committed:
Sustainable InnerSource donut pattern (#453)

Sustainable InnerSource Program - donut pattern

---------

Co-authored-by: Chan Voong <Chan_Voong@comcast.com>
Co-authored-by: Chan Voong <34136874+voongc@users.noreply.github.com>

zilio committed:
Add files for mindmap in pt-br (#620)

spier committed:
Test regeneration of pt-br mindmap (#621)

* Updating CODEOWNERS file
* Writing updated toc.md for the pt-br book
* Re-creating markmap and screenshot

sicot-f committed:
Airbus pattern use (#622)

Add Airbus as a "known instance" to multiple patterns

gruetter committed:
InnerSource License: Added Robert Bosch GmbH as known instance (#532)

* Added Robert Bosch GmbH as Known Instance
* Add another force related to possible relicensing when moved from InnerSource to open source
* Reformatting pattern, to include Airbus reference (pulled in via last update from main)

---------

Co-authored-by: Sebastian Spier <github@spier.hu>

spier committed:
Adding Patterns WG board report 2023-11 (#626)

* Adding Patterns WG board report 2023-11

spier committed:
Move files from project-setup to standard folders (#633)

* Move files from project-setup to standard folders
* Fix all links from patterns to project-setup folder
* Fix all links from translations to project-setup folder
* Remove instances of project/setup in book and scripts
* Point from old project-setup folder to the new folder for these files
* Regenerate en toc.me locally. Not sure why the GHA did not do this.

gruetter committed:
improved review committee pattern #634 (#635)

rrrutledge committed:
Add wiki implementation of InnerSource portal pattern (#637)

Add wiki implementation of InnerSource portal pattern. To structure the information more clearly, this adds a new "Implementation" section to this pattern, which complements the "Solution" and "Known Instances" section.

rrrutledge committed:
Fix Markdown Lint error (#644)

Add alt text

https://github.com/InnerSourceCommons/InnerSourcePatterns/actions/runs/7386496303/job/20093212394

spier committed:
Fix wording in transitioning-contractor-code-to-innersource-model.md (#653)

rmarting committed:
:twisted_rightwards_arrows: Merge latest changes

rmarting committed:
:twisted_rightwards_arrows: Fix merge conflict
